vcd/datasource_vcd_nsxv_firewall.go

```diff
@@ -0,0 +1,221 @@
+package vcd
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func datasourceVcdNsxvFirewall() *schema.Resource {
+	return &schema.Resource{
+		Read: resourceVcdNsxvFirewallRead,
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway name in which the firewall rule is located",
+			},
+			"rule_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Firewall rule ID for lookup",
+			},
+			"name": &schema.Schema{
```

Comment:
We can't use name, cause it isn't unique?

Reply:
`name` is computed